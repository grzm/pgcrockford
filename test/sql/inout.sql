SELECT b32
  FROM (VALUES ('0'::crockford.crockford2),
               ('-0-'),
               ('-1'),
               ('-ololi-'),
               ('1ZZZ')) AS _ (b32);


SELECT '2000'::crockford.crockford2;
SELECT '-'::crockford.crockford2;
SELECT '+0'::crockford.crockford2;
SELECT 'U'::crockford.crockford2;
SELECT 'u'::crockford.crockford2;

SELECT 65535::crockford.crockford2;
SELECT 65536::crockford.crockford2;

SELECT b32
  FROM (VALUES ('0'::crockford.crockford4),
               ('-0-'),
               ('-1'),
               ('-ololi-'),
               ('3ZZZZZZ')) AS _ (b32);

SELECT 4294967295::crockford.crockford4;
SELECT 4294967296::crockford.crockford4;

SELECT '4000000'::crockford.crockford4;
SELECT '-'::crockford.crockford4;
SELECT '+0'::crockford.crockford4;

SELECT b32
  FROM (VALUES ('0'::crockford.crockford8),
               ('-0-'),
               ('-1'),
               ('-ololi-'),
               ('FZZZZZZZZZZZZ')) AS _ (b32);

SELECT 'G000000000000'::crockford.crockford8;
SELECT '-'::crockford.crockford8;
SELECT '+0'::crockford.crockford8;

