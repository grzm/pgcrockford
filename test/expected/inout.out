SELECT b32
  FROM (VALUES ('0'::crockford.crockford2),
               ('-0-'),
               ('-1'),
               ('-ololi-'),
               ('1ZZZ')) AS _ (b32);
 b32  
------
 0
 0
 1
 1011
 1ZZZ
(5 rows)

SELECT '2000'::crockford.crockford2;
ERROR:  value "2000" is out of range for type crockfordint2
LINE 1: SELECT '2000'::crockford.crockford2;
               ^
SELECT '-'::crockford.crockford2;
ERROR:  invalid input syntax for crockford value: "-"
LINE 1: SELECT '-'::crockford.crockford2;
               ^
SELECT '+0'::crockford.crockford2;
ERROR:  invalid input syntax for crockford value: "+0"
LINE 1: SELECT '+0'::crockford.crockford2;
               ^
DETAIL:  symbol "+" is invalid
SELECT 'U'::crockford.crockford2;
ERROR:  invalid input syntax for crockford value: "U"
LINE 1: SELECT 'U'::crockford.crockford2;
               ^
DETAIL:  symbol "U" is invalid
SELECT 'u'::crockford.crockford2;
ERROR:  invalid input syntax for crockford value: "u"
LINE 1: SELECT 'u'::crockford.crockford2;
               ^
DETAIL:  symbol "u" is invalid
SELECT 65535::crockford.crockford2;
 crockford2 
------------
 1ZZZ
(1 row)

SELECT 65536::crockford.crockford2;
ERROR:  crockford2 out of range
SELECT b32
  FROM (VALUES ('0'::crockford.crockford4),
               ('-0-'),
               ('-1'),
               ('-ololi-'),
               ('3ZZZZZZ')) AS _ (b32);
   b32   
---------
 0
 0
 1
 1011
 3ZZZZZZ
(5 rows)

SELECT 4294967295::crockford.crockford4;
 crockford4 
------------
 3ZZZZZZ
(1 row)

SELECT 4294967296::crockford.crockford4;
ERROR:  crockford4 out of range
SELECT '4000000'::crockford.crockford4;
ERROR:  value "4000000" is out of range for type crockfordint4
LINE 1: SELECT '4000000'::crockford.crockford4;
               ^
SELECT '-'::crockford.crockford4;
ERROR:  invalid input syntax for crockford value: "-"
LINE 1: SELECT '-'::crockford.crockford4;
               ^
SELECT '+0'::crockford.crockford4;
ERROR:  invalid input syntax for crockford value: "+0"
LINE 1: SELECT '+0'::crockford.crockford4;
               ^
DETAIL:  symbol "+" is invalid
SELECT b32
  FROM (VALUES ('0'::crockford.crockford8),
               ('-0-'),
               ('-1'),
               ('-ololi-'),
               ('FZZZZZZZZZZZZ')) AS _ (b32);
      b32      
---------------
 0
 0
 1
 1011
 FZZZZZZZZZZZZ
(5 rows)

SELECT 'G000000000000'::crockford.crockford8;
ERROR:  value "G000000000000" is out of range for type crockfordint8
LINE 1: SELECT 'G000000000000'::crockford.crockford8;
               ^
SELECT '-'::crockford.crockford8;
ERROR:  invalid input syntax for crockford value: "-"
LINE 1: SELECT '-'::crockford.crockford8;
               ^
SELECT '+0'::crockford.crockford8;
ERROR:  invalid input syntax for crockford value: "+0"
LINE 1: SELECT '+0'::crockford.crockford8;
               ^
DETAIL:  symbol "+" is invalid
